### PR TITLE
LUCENE-9325: Make Sort final

### PR DESCRIPTION
**Pull request by romseygeek**
Sort is used in all sorts of settings where we assume that it is immutable
(for example, in IndexWriterConfig).  This commit makes it so, plus it also
updates the severely outdated javadoc.

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/search/Sort.java [hunk 1]
```diff
@@ -108,20 +55,6 @@ public Sort() {
 
   /** Sorts by the criteria in the given SortField. */
   public Sort(SortField field) {
```
Comment (danmuzi) [inline lucene/core/src/java/org/apache/lucene/search/Sort.java:57]: By changing the method below to a constructor, this constructor seems to be no longer needed.
Our new variable arguments constructor can cover it.
Even without this, all the following cases can be covered.
```java
new Sort(); // Pass
new Sort(new SortField("fld", SortField.Type.STRING)); // Pass
new Sort(new SortField[]{new SortField("fld", SortField.Type.STRING), new SortField("fld2", SortField.Type.INT)}); // Pass
new Sort(new SortField("fld", SortField.Type.STRING), new SortField("fld2", SortField.Type.INT)); // Pass
new Sort(new SortField[0]); // IllegalArgumentException
```

**Review by danmuzi — COMMENTED** [review]
This patch looks good to me! @romseygeek 
I left a simple comment.

**Review by danmuzi — APPROVED** [review]
Thank you for checking!

File: lucene/CHANGES.txt [hunk 2]
```diff
@@ -153,7 +153,9 @@ API Changes
 * LUCENE-10115: Add an extension point, BaseQueryParser#getFuzzyDistance, to allow custom
   query parsers to determine the similarity distance for fuzzy queries. (Chris Hegarty)
 
-*  LUCENE-10132: Support addition of diagnostics by custom merge policies (Chris Hegarty)
+* LUCENE-10132: Support addition of diagnostics by custom merge policies (Chris Hegarty)
+
+* LUCENE-9325: Sort is now final, and the `setSort()` method has been removed
```
Comment (danmuzi) [inline lucene/CHANGES.txt:158]: I think the name is missing.

Comment (romseygeek) [reply lucene/CHANGES.txt]: Whoops, good spot!

**Pull request merged**